File: backend/models/travel.py
"""Domain models shared by the LLM extraction layer and the HTTP API."""

from typing import Literal

from pydantic import BaseModel, Field

TravelCategory = Literal[
    "beach",
    "cold",
    "city",
    "adventure",
    "culture",
    "nature",
]

BudgetLevel = Literal["low", "medium", "high"]

Month = Literal[
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
]


class TravelPreferences(BaseModel):
    """Structured output of the LLM layer. Every field is optional because a
    single spoken sentence rarely mentions all of them."""

    destination: str | None = None
    category: TravelCategory | None = None
    month: Month | None = None
    travelers: int | None = None
    budget_level: BudgetLevel | None = None
    max_budget: float | None = None


class TravelPackage(BaseModel):
    id: int
    name: str
    destination: str
    country: str
    category: str
    description: str
    days: int
    price: float
    max_people: int
    best_months: list[str]


class Recommendation(TravelPackage):
    """A package plus why the search layer ranked it where it did."""

    score: int
    match_reasons: list[str]


class RecommendationRequest(BaseModel):
    text: str = Field(min_length=1, max_length=1000)


class RecommendationResponse(BaseModel):
    preferences: TravelPreferences
    recommendations: list[Recommendation]

File: frontend/types/travel.ts
/** Mirrors the Pydantic models in backend/models/travel.py. */

export type TravelCategory =
  "beach" | "cold" | "city" | "adventure" | "culture" | "nature";

export type BudgetLevel = "low" | "medium" | "high";

export type Month =
  | "January"
  | "February"
  | "March"
  | "April"
  | "May"
  | "June"
  | "July"
  | "August"
  | "September"
  | "October"
  | "November"
  | "December";

export interface TravelPreferences {
  destination: string | null;
  country: string | null;
  category: TravelCategory | null;
  month: Month | null;
  travelers: number | null;
  budget_level: BudgetLevel | null;
  max_budget: number | null;
}

export interface Recommendation {
  id: number;
  name: string;
  destination: string;
  country: string;
  category: string;
  description: string;
  days: number;
  price: number;
  max_people: number;
  best_months: string[];
  score: number;
  match_reasons: string[];
}

export interface RecommendationResponse {
  preferences: TravelPreferences;
  recommendations: Recommendation[];
}
